turbopack/crates/turbo-tasks/src/effect.rs

```diff
@@ -0,0 +1,109 @@
+use std::{borrow::Cow, future::Future, panic, pin::Pin};
+
+use anyhow::{anyhow, Result};
+use auto_hash_map::AutoSet;
+use parking_lot::Mutex;
+use tokio::task::JoinHandle;
+use tracing::{Instrument, Span};
+
+use crate::{self as turbo_tasks, emit, manager::turbo_tasks_future_scope, CollectiblesSource, Vc};
+
+#[turbo_tasks::value_trait]
+trait Effect {}
```

Comment:
Needs some documentation about why somebody might use this. The documentation should also link to `EffectInstance`.